### PR TITLE
add support for {Global,Command}Args in CreateContainer

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind feature
#### What this PR does / why we need it:
This should shore up most of the remaining pieces podman needs for container creation

#### Does this PR introduce a user-facing change?

```release-note
Add support for RuntimeArgs and RuntimeOpts in CreateContainer RPC
```

### CODE REVIEW

**Comment by haircommander** [comment]
cc @mheon

**Comment by mheon** [comment]
This should do everything we're looking for - thanks!

File: conmon-rs/server/src/rpc.rs [hunk 1]
```diff
@@ -87,16 +108,12 @@ impl conmon::Server for Server {
...
         let runtime = self.config().runtime().clone();
-        let exit_paths: Vec<PathBuf> = pry!(pry!(req.get_exit_paths())
-            .iter()
-            .map(|r| r.map(PathBuf::from))
-            .collect());
-        let oom_exit_paths: Vec<PathBuf> = pry!(pry!(req.get_oom_exit_paths())
-            .iter()
-            .map(|r| r.map(PathBuf::from))
-            .collect());
+        let exit_paths: Vec<PathBuf> = capnp_vec_path!(req.get_exit_paths());
```
Comment (saschagrunert) [inline conmon-rs/server/src/rpc.rs:115]:
```suggestion
        let exit_paths = capnp_vec_path!(req.get_exit_paths());
```

File: conmon-rs/server/src/rpc.rs [hunk 2]
```diff
@@ -87,16 +108,12 @@ impl conmon::Server for Server {
...
+        let oom_exit_paths: Vec<PathBuf> = capnp_vec_path!(req.get_oom_exit_paths());
```
Comment (saschagrunert) [inline conmon-rs/server/src/rpc.rs:116]:
```suggestion
        let oom_exit_paths = capnp_vec_path!(req.get_oom_exit_paths());
```

File: conmon-rs/server/src/server.rs [hunk 3]
```diff
@@ -239,6 +241,10 @@ impl Server {
             args.push(Self::SYSTEMD_CGROUP_ARG.into());
         }
 
+        if runtime_args.len() > 0 {
+            args.extend(runtime_args);
+        }
```
Comment (saschagrunert) [inline conmon-rs/server/src/server.rs:246]: I think we do not have to pre-check the length:
```suggestion
        args.extend(runtime_args);
```

File: pkg/client/client.go [hunk 4]
```diff
@@ -486,6 +486,14 @@ type CreateContainerConfig struct {
...
 	// CleanupCmd is the command that will be executed once the container exits
 	CleanupCmd []string
+
+	// RuntimeArgs are the additional arguments passed to the create runtime call
+	// before the command. e.g: crun --runtime-arg create
+	RuntimeArgs []string
+
+	// RuntimeOpts are the additional arguments passed to the create runtime call
+	// after the command. e.g: crun create --runtime-opt
+	RuntimeOpts []string
```
Comment (saschagrunert) [inline pkg/client/client.go:496]: I'm not sure about the naming args vs opts. How about having something like:
```go
type Arguments struct {
	// Global options to be added to the runtime call.
	Global  []string

	// Command options added to the runtime call.
	Command []string
}
```

Beside that, runc states that `global options` are our `RuntimeArgs` while `arguments` are our defined `RuntimeOpts`:
```
USAGE:
   runc [global options] command [command options] [arguments...]
```

Comment (haircommander) [reply pkg/client/client.go:496]: good idea! I agree it was hard to interpret. I have changed to `GlobalArgs` and `CommandArgs`

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**